src/components/ImportAddressesButton.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { 
+  bitcoindImportMulti,
+  bitcoindGetAddressStatus,
+  bitcoindParams,
+} from '../bitcoind';
+
+// Components
+import { FormHelperText, Button, Box, Switch, FormControlLabel, Tooltip, makeStyles } from '@material-ui/core'
+import InfoIcon from '@material-ui/icons/Info';
+
+const useStyles = makeStyles(() => ({
+  tooltip: {
+    fontSize: '.8rem'
+  }
+}))
+/**
+ * @description A button for importing a set of given addresses. Includes a switch
+ * to optionally choose to rescan the blockchain for balances.
+ */
+function ImportAddressesButton({ 
+  addresses = [], 
+  client,
+  importCallback 
+}) {
+  const [imported, setImported] = useState(false)
+  const [importError, setImportError] = useState('fooey! import error')
+  const [rescan, setRescanPreference] = useState(false)
+  const [addressesError, setAddressesError] = useState('')
+  const [enableImport, setEnableImport] = useState(false)
+
+  const classes = useStyles()
+  // when addresses prop has changed, we want to check its status
+  // Address management currently isn't optimized and so they are added to the store
+  // one at at time. This effect is run each time that updates. Eventually,
+  // this should be optimized to happen all at once. For now though, once we find
+  // a single address that can be imported then enable the import button and 
+  // we won't run this anymore. This is mostly for wallet view, not script entry
+  useEffect(() => {
+    const checkAddress = async () => {
+      // reset in case we can't import
+      setEnableImport(false)
+      const address = addresses[addresses.length - 1] // TODO: loop, or maybe just check one
```

Comment:
This comment is from the old component. As noted in the comments above the `useEffect` call, there are several ways this could be optimized, most notably, the slices shouldn't be added to the store as individual actions since each update causes re-renders in the app, which has side-effects to make other network calls throughout. But for now it felt like something to hold off on for later as this update was already getting pretty big.